crates/uv-interpreter/src/find_python.rs

```diff
@@ -43,16 +43,35 @@ pub fn find_requested_python(request: &str, cache: &Cache) -> Result<Option<Inte
             _ => unreachable!(),
         };
         find_python(selector, cache)
-    } else if !request.contains(std::path::MAIN_SEPARATOR) {
-        // `-p python3.10`; Generally not used on windows because all Python are `python.exe`.
-        let Some(executable) = find_executable(request)? else {
-            return Ok(None);
-        };
-        Interpreter::query(executable, cache).map(Some)
     } else {
-        // `-p /home/ferris/.local/bin/python3.10`
-        let executable = uv_fs::absolutize_path(request.as_ref())?;
-        Interpreter::query(executable, cache).map(Some)
+        match fs_err::metadata(request) {
+            Ok(metadata) => {
+                // Map from user-provided path to an executable.
+                let path = uv_fs::absolutize_path(request.as_ref())?;
+                let executable = if metadata.is_dir() {
+                    // If the user provided a directory, assume it's a virtual environment.
+                    // `-p /home/ferris/.venv`
+                    if cfg!(windows) {
+                        Cow::Owned(path.join("Scripts/python.exe"))
+                    } else {
+                        Cow::Owned(path.join("bin/python"))
+                    }
```

Comment:
I don't personally care, but this probably won't handle environments like cygwin or mingw64, as I think they use a `bin` directory, because they are emulating Unix. They will presumably still have a `.exe` suffix, though, as that's a Windows requirement.

I suspect it's better to ignore this problem until someone using one of those environments speaks up, because they will be able to give better information than my guesses...

Reply:
Yeah makes total sense. I think we _already_ have this problem in a few places so I'm okay with using the same pattern until we fix it holistically...